src/borg/archiver/prune_cmd.py

```diff
@@ -18,27 +26,33 @@
 logger = create_logger()
 
 
-def prune_within(archives, seconds, kept_because):
-    target = datetime.now(timezone.utc) - timedelta(seconds=seconds)
-    kept_counter = 0
-    result = []
-    for a in archives:
-        if a.ts > target:
-            kept_counter += 1
-            kept_because[a.id] = ("within", kept_counter)
-            result.append(a)
-    return result
+# The *_period_func group of functions create period grouping keys to group together archives falling within a certain
+# period. Among archives in each of these groups, only the latest (by creation timestamp) is kept.
+
+
+def unique_period_func():
+    counter = 0
+
+    def inner(_a):
+        """Group archives by an incrementing counter, practically making each archive a group of 1"""
+        nonlocal counter
+        counter += 1
+        return counter
+
+    return inner
```

Comment:
Would something like this work and be clearer?  It relies on the fact that, by default, objects of a class are not equal to anything else (or each other).  (Code untested.)
```
class NeverEqual(): pass

def unique_values(_a):
    return NeverEqual()

def unique_period_func():
    return unique_values
```

Reply:
I think keeping them numbers makes for easier textual debugging if this code ever needs triage. However, maybe there is a sequence generator that I can use instead that communicates more clearly what these numbers mean.

Reply:
Replaced the naive counter with one from `itertools` and renamed the inner function to describe the values it returns. I think this has improved readability and is a little more intuitive than relying on the fact that objects are unequal by default (which was not immediately obvious to me, I am not very well versed in Python).

Reply:
I think the itertools method is good.